koku/api/resource_types/aws_ec2_compute_instances/view.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright 2024 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""View for AWS accounts."""
+from django.conf import settings
+from django.db.models import F
+from django.db.models.functions import Coalesce
+from django.utils.decorators import method_decorator
+from django.views.decorators.vary import vary_on_headers
+from rest_framework import filters
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.response import Response
+
+from api.common import CACHE_RH_IDENTITY_HEADER
+from api.common.pagination import ResourceTypeViewPaginator
+from api.common.permissions.aws_access import AwsAccessPermission
+from api.resource_types.serializers import ResourceTypeSerializer
+from reporting.provider.aws.models import AWSCostEntryLineItemSummaryByEC2Compute
+
+
+class AWSEC2ComputeInstanceView(generics.ListAPIView):
+    """API GET list view for AWS EC2 compute instances."""
+
+    queryset = (
+        AWSCostEntryLineItemSummaryByEC2Compute.objects.annotate(
+            **(
+                {
+                    "value": F("resource_id"),
+                    "alias": Coalesce(F("instance_name"), "resource_id"),
+                }
+            )
+        )
+        .values("value", "alias")
+        .distinct()
+    )
+
+    serializer_class = ResourceTypeSerializer
+    permission_classes = [AwsAccessPermission]
+    filter_backends = [filters.OrderingFilter, filters.SearchFilter]
+    ordering = ["value", "alias"]
+    search_fields = ["value", "alias"]
+    pagination_class = ResourceTypeViewPaginator
+
+    @method_decorator(vary_on_headers(CACHE_RH_IDENTITY_HEADER))
+    def list(self, request):
+        # Reads the aws ec2 compute instances info and displays values related to what the user has access to.
+        supported_query_params = ["search", "limit"]
+        user_access = []
+        error_message = {}
+        # Test for only supported query_params
+        if self.request.query_params:
+            for key in self.request.query_params:
+                if key not in supported_query_params:
+                    error_message[key] = [{"Unsupported parameter"}]
+                    return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This looks like the view is doing input validation. The [serializer](https://www.django-rest-framework.org/api-guide/serializers/) should handle [validation](https://www.django-rest-framework.org/api-guide/serializers/#validation) not the view.

Reply:
I don't disagree, but I think moving this validation to the serializer will require an over haul of all the resource_type views as they all have this and rely on the same serializer class. See #3039